src/client/components/series-menu/series-menu.tsx

```diff
@@ -34,18 +36,17 @@ interface SeriesMenuProps {
   openOn: Element;
   containerStage: Stage;
   onClose: Fn;
-  // TODO: fix
-  series: any;
+  series: Series;
   inside?: Element;
 }
 
 interface SeriesMenuState {
-  format: SeriesFormat;
+  series: Series;
 }
 
 export class SeriesMenu extends React.Component<SeriesMenuProps, SeriesMenuState> {
 
-  state: SeriesMenuState = { format: this.props.series.format };
+  state: SeriesMenuState = { series: this.props.series };
```

Comment:
this prop shouldn't change after creating that component - we need to enforce that with key prop at this component

Reply:
And series should implement something like `idKey :: string`